### PR TITLE
SDKv2 to Framework migration guide

### PR DESCRIPTION
**Pull request by johnsonaj**
### Output from Acceptance Testing

```console
% make testacc TESTS=TestAccXXX PKG=ec2

...
```

### CODE REVIEW

File: docs/migrate-from-sdk-to-framework.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+Convert a resource:
+
+The following pattern is used to generate a file:  `tfsdk2fw [-resource <resource-type>|-data-source <data-source-type>] <package-name> <name> <generated-file>`
+
+Example:
+
+```console
+tfsdk2fw -resource aws_example_resource examplepackage ResourceName internal/service/examplepackage/resource_name_fw.go
```
Comment (jar-b) [inline docs/migrate-from-sdk-to-framework.md:22]: Does this create a separate resource, and if so, what is the lifecycle for the new file/resource? Will it live alongside the Plugin SDK V2 resource for some time (testing) and then be replaced?

**Review by jar-b — COMMENTED** [review]
Looks great. Just a couple suggestions and a clarification on the output of `tfsdk2fw`.

**Review by gbw — COMMENTED** [review]
Thanks. This is really helpful. Two things I've noticed when using framework are the custom types don't play well with autoflex because it reflects to the wrong type and attributes which are computed where the provider can return nil requires use of legacy functions. I'm unsure if both of these are bugs and/or it makes sense to add here or some other doc about the use of Plugin Framework for AWS.

File: docs/migrate-from-sdk-to-framework.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+## State Upgrade
+
+Terraform Plugin Framework introduced `null` values, which differ from `zero` values. Since the Plugin SDKv2 marked both `null` and `zero` values as the same, it will be necessary to use the [State Upgrader](https://developer.hashicorp.com/terraform/plugin/framework/migrating/resources/state-upgrade).
```
[the remark this answers is not in the document]
Comment (johnsonaj) [reply docs/migrate-from-sdk-to-framework.md:27]: We do. I did one recently that uses a state upgrader to migrate `string` to a custom `ARN`. I can add a link to that file and a code snippet

**Pull request merged**